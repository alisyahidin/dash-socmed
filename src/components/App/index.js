import React, { Component } from 'react'
import {
  BrowserRouter as Router,
  Route,
  Switch,
  Redirect
} from 'react-router-dom'

import Login from '../../containers/Login/'
import Home from '../../containers/Home/'
import Profile from '../../containers/Profile/'
import Users from '../../containers/Users/'
import NotFound from '../../containers/NotFound/'

import storage from '../../lib/storage'

class App extends Component {
  render() {
    return (
      <Router>
        <Switch>
          <Route exact path="/" component={Home} />
          <Route exact path="/users" component={Users} />
          <Route path="/users/:id" component={Profile} />
          <Route
            path="/profile"
            render={props => storage.has('user') ? <Profile {...props} /> : <Redirect to="/login" />}
          />
          <Route
            path="/login"
            render={props => !storage.has('user') ? <Login {...props} /> : <Redirect to="/" />}
          />
          <Route component={NotFound} />
        </Switch>
      </Router>
    )
  }
}

export default App
