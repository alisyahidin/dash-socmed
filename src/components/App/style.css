.main-content {
  margin: 0.5rem 0rem;
}