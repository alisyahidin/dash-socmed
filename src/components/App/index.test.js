import React from 'react'
import { shallow } from 'enzyme'
import { Route } from 'react-router-dom'
import App from './index'

it('render <App /> correctly', () => {
  const component = shallow(<App />)

  expect(component).toMatchSnapshot()
})

it('count valid Route components', () => {
  const component = shallow(<App />)

  expect(component.find(Route)).toHaveLength(6)
})