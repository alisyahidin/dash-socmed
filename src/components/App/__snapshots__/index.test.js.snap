// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <App /> correctly 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/users"
    />
    <Route
      component={[Function]}
      path="/users/:id"
    />
    <Route
      path="/profile"
      render={[Function]}
    />
    <Route
      path="/login"
      render={[Function]}
    />
    <Route
      component={[Function]}
    />
  </Switch>
</BrowserRouter>
`;
